### DISCUSSION TITLE
I want to contribute in the development of Melonds

### DISCUSSION THREAD

**Discussion by HarisOfNazareth** [rafaelvcaetano/melonDS-android#1073] [Q&A]
The thing is that i want to contribute in the development of the Melonds. I am just a newbie programmer btw. **Android SDK, NDK and CMake**, Are these the only libraries we need to develop Melonds? I am asking so that i come to help later when i master these languages about in a year or so.

**Comment by rafaelvcaetano** [comment]
First of all, I love your enthusiasm! 

Now, since you are learning, everything you described are tools and not libraries nor languages. But yes, all of that is used in the development of the emulator. Since you are starting to learn, and if you want to contribute to melonDS in the long run, you will need to decide on what you want to focus on.

Is it the emulator itself? If so, you will need to learn the following:
* C++ programming language
* CMake
* Assembly (if you want to dab a bit on the JIT compiler)

If you want to work on the frontend, you will need to learn the following:
* Kotlin programming language
* Android SDK
* A bit of C++

If you want to help with everything, well, you will need to learn all of the above. This won't be easy, but hey, I hope to see your contributions in a year! I would like to help you more, but I don't have a lot of free time. However, if you want some tips or pointers, just let me know!

**Comment by HarisOfNazareth** [comment]
Thank you so much Sir. I will try my hardest to master programming and I
won't let you down.
